sandbox/grist/imports/import_csv.py

```diff
@@ -90,108 +90,116 @@ def parse_file(file_path, parse_options=None):
   """
   parse_options = parse_options or {}
 
-  with open(file_path, "rb") as f:
+  with codecs.open(file_path, "rb") as f:
+    sample = f.read(100000)
+  encoding = chardet.detect(sample)['encoding'] or "utf8"
+  # In addition, always prefer UTF8 over ASCII.
+  if encoding == 'ascii':
+    encoding = 'utf8'
+  log.info("Using encoding %s" % encoding)
+
+  with codecs.open(file_path, mode="r", encoding=encoding) as f:
     parsing_options, export_list = _parse_open_file(f, parse_options=parse_options)
     return parsing_options, export_list
 
 
+def _guess_dialect(file_obj):
+  try:
+    # Restrict allowed delimiters to prevent guessing other char than this list.
+    dialect = csv.Sniffer().sniff(file_obj.read(100000), delimiters=['\t', ',', ';', '|'])
+    log.info("Guessed dialect %s" % dict(dialect.__dict__))
```

Comment:
```suggestion
    log.info("Guessed dialect %s", dict(dialect.__dict__))
```

Why the `dict`?

Reply:
Otherwise we just have the `repr` which is useless imho (something like `<Dialect xxx>`).